### PR TITLE
improve ditto weatherboost handling

### PR DESCRIPTION
**Pull request by Fabio1988**

### CODE REVIEW

File: decoder/pokemon.go [hunk 1]
```diff
@@ -740,6 +755,37 @@ func (pokemon *Pokemon) updatePokemonFromDiskEncounterProto(ctx context.Context,
...
+		pokemon.PokemonId = pokemonId
+	}
+}
+
+func (pokemon *Pokemon) setWeather(weather null.Int) {
+	if pokemon.PokemonId != 0 && pokemon.Weather.ValueOrZero() != weather.ValueOrZero() {
+		if pokemon.IsDitto {
+			if weather.ValueOrZero() == 3 {
+				// both Ditto and disguise are boosted and Ditto was not boosted: none -> boosted
+				//pokemon.Level += null.IntFrom(5)
```
Comment (Fabio1988) [inline decoder/pokemon.go:773]: is there an easy way to increment the value of a null.Int prop? :)

Comment (jfberry) [reply decoder/pokemon.go:773]: Can you do

pokemonLevel.Value += 5
?
Does that work?

Comment (Fabio1988) [reply decoder/pokemon.go]: that did not work, but I pushed a possibility where the compiler did not complain

Comment (jfberry) [reply decoder/pokemon.go]: I misremembered the struct member names. That is what I meant and I think it’s fine as the struct just contains two bits

**Comment by Fabio1988** [comment]
running this since yesterday.. unfortunately no weather-boost for dittos yet, maybe today :)

**Comment by Fabio1988** [comment]
image
my first reset of IV from weatherboosted ditto

**Comment by Mygod** [comment]
Maybe I will start a PR...

**Comment by Fabio1988** [comment]
I would appreciate it.. the workflow is bit different and I tried to cover as much as possible from Chuck.

File: decoder/pokemon.go [hunk 2]
```diff
@@ -368,20 +378,22 @@ func (pokemon *Pokemon) updateFromWild(ctx context.Context, db db.DbDetails, wil
...
 	encounterId := strconv.FormatUint(wildPokemon.EncounterId, 10)
 	oldWeather, oldPokemonId := pokemon.Weather, pokemon.PokemonId
+	pokemonId := int16(wildPokemon.Pokemon.PokemonId)
+	pokemon.setPokemonDisplay(pokemonId, wildPokemon.Pokemon.PokemonDisplay)
 
 	if pokemon.Id == "" || pokemon.SeenType.ValueOrZero() == SeenType_Cell || pokemon.SeenType.ValueOrZero() == SeenType_NearbyStop {
 		updateStats(ctx, db, encounterId, stats_seenWild)
 	}
+	if pokemon.IsDitto {
+		// TODO: add some more logic regarding event change, actually a Ditto will remain as Ditto
```
Comment (Mygod) [inline decoder/pokemon.go:388]: What do you mean? Source?

Comment (Fabio1988) [reply decoder/pokemon.go:388]: When Spotlight hour starts that ditto would probably change to the spotlight hour mon, false? Would it stay a ditto for sure?

Comment (Mygod) [reply decoder/pokemon.go:388]: I am guessing that it should completely reroll. Haven't tested this though.

Comment (Mygod) [reply decoder/pokemon.go:388]: Guess what. Using #65, I caught a Snubbull disguised as a Ditto rerolled into a Houndour (impossible to be Ditto).

```
INFO 2023-03-23 03:52:29 Pokemon 16593934753357378148 changed from (209,1253,0,1) to (228,229,0,1)
INFO 2023-03-23 03:52:29 [POKEMON] 16593934753357378148: B0>00/PP Ditto found, disguised as 228. (4,-4,0/0/0,148)>(0,1,4/9/0)
```

Comment (Fabio1988) [reply decoder/pokemon.go:388]: a/b spawn I would say :/

**Pull request merged**